src/main/java/ru/hogwarts/school/controller/StudentController.java

```diff
@@ -47,10 +48,25 @@ public ResponseEntity<Void> deleteStudent(@RequestParam Long id) {
     }
 
     @GetMapping
-    public ResponseEntity<Collection<Student>> getStudentsCertainAge(@RequestParam(required = false) int age) {
+    public ResponseEntity<Collection<Student>> getStudentsCertainAge(
+            @RequestParam(defaultValue = "0") int age,
+            @RequestParam(defaultValue = "0") int minAge,
+            @RequestParam(defaultValue = "0") int maxAge) {
         if (age > 0) {
-            return ResponseEntity.ok(studentService.findStudentByAge(age));
+            return ResponseEntity.ok(studentService.findStudentsByAge(age));
+        }
+        if (minAge > 0 && maxAge > minAge) {
+            return ResponseEntity.ok(studentService.findStudentsByAgeBetween(minAge, maxAge));
         }
         return ResponseEntity.ok(Collections.emptyList());
     }
+
+    @GetMapping("/getFaculty")
+    public ResponseEntity<Faculty> getStudentFaculty(@RequestParam Long id) {
```

Comment:
Неправильно запрос сделан с точки зрения правил REST. Во-первых, надо обратиться по id к конкретному ресурсу. Во-вторых, надо получить его свойство

```suggestion
    @GetMapping("/{id}/faculty")
    public ResponseEntity<Faculty> getStudentFaculty(@PathVariable Long id) {
```